### PR TITLE
Add a recommendation to turn on type hints with untyped declaration warning

### LINKED ISSUE #82587: Default Node GDScript Template has Warnings/Errors with new Untyped Declaration Warning
### Godot version

v4.2.dev5.official [e3e2528ba]

### System information

Godot v4.2.dev5 - Windows 10.0.19045 - Vulkan (Mobile) - dedicated NVIDIA GeForce GTX 980 Ti (NVIDIA; 31.0.15.3699) - Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 Threads)

### Issue description

When creating a new GDScript with the default Node template, untyped declarations are used:
```
extends Node3D


# Called when the node enters the scene tree for the first time.
func _ready():
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass
```
I believe this is a bug because a template that has untyped declarations will cause warnings or errors if the new "[Untyped Declaration](https://github.com/godotengine/godot/pull/81355)" warning is enabled in the project settings. Additionally, the [default template described in the documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/creating_script_templates.html#default-template) uses typed declarations and does not match the default template currently used by the editor.

### Steps to reproduce

1. Ensure that no [Editor-defined templates](https://docs.godotengine.org/en/latest/tutorials/scripting/creating_script_templates.html#editor-defined-templates) exist
2. Create a new project
3. Under Project Settings (Advanced Settings) -> Debug -> GDScript -> Untyped Declaration: Set to `Warn` or `Error`
4. Create a new GDScript through the FileSystem panel or by choosing Attach Script on a new scene's root node
5. Note that the new GDScript uses untyped declarations and does not match the [documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/creating_script_templates.html#default-template)

image
image

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by allenwp**
Fixes #82587 by improving usability of the untyped_declaration warning. This adds a note to the documentation that recommends turning on EditorSettings.text_editor/completion/add_type_hints when the untyped declaration warning is set to warn or error. This new text echos the recommendation proposed in the [docs PR](https://github.com/godotengine/godot-docs/pull/8130)

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -555,7 +555,7 @@
 			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when returning a call from a [code]void[/code] function when such call cannot be guaranteed to be also [code]void[/code].
 		</member>
 		<member name="debug/gdscript/warnings/untyped_declaration" type="int" setter="" getter="" default="0">
-			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a variable or parameter has no static type, or if a function has no static return type.
+			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a variable or parameter has no static type, or if a function has no static return type. It is recommended to turn on [member EditorSettings.text_editor/completion/add_type_hints] when this is set to [code]warn[/code] or [code]error[/code].
```
Comment (YuriSizov) [inline doc/classes/ProjectSettings.xml:558]:
```suggestion
			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when a variable or parameter has no static type, or if a function has no static return type.
			[b]Note:[/b] This warning is recommended together with [member EditorSettings.text_editor/completion/add_type_hints] to help achieve type safety.
```

**Review by YuriSizov — APPROVED** [review]
I would reword this slightly to better fit our overall style and formatting. But should be good otherwise.

**Comment by allenwp** [comment]
Suggested change looks like an improvement to me!

image
image

I've squashed the commits, so I think they should be ready for merge.

**Comment by akien-mga** [comment]
Thanks!
